### DISCUSSION TITLE
help with kata in js... 6 pass 4 fail??

### DISCUSSION THREAD

**Discussion by Iansogotthis** [codewars/codewars.com#3148] [Q&A]
WHAT IS WRONG WITH MY LOGIC
![kata2](https://github.com/codewars/codewars.com/assets/83776648/2bafcc3d-2601-4e44-8595-6db66ca3887f)

```
function validISBN10(isbn) {
  let isbnArray = isbn.split("")
  if(isbnArray.includes("X")){
    isbnArray.pop()
    isbnArray.push(10)

    
  }

  if(isbnArray.length !== 9){
    return false
  }
  
  else{
    for(let i = 0; i < isbn.length; i+=1){
      
      let count = 0

      if(Number.isInteger("i")&& (i <= 10 && i <= 0)){
        count += i;
      }
      else{
      return false
      }
    }
  
    if(count % 11 === 0){
    return true
    }
    
  }
//console.log(isbnArray)
}
```
My error log is in the picture
link to challenge: https://www.codewars.com/kata/51fc12de24a9d8cb0e000001/train/javascript

**Comment by hobovsky** [comment]
Codewars Github is meant as a place to discuss functions and issues with Codewars website and platform. It is not meant for help with solving kata. Please ask your question in kata discourse (label your post as `question`), or come for help to `#help-solve` channel of Codewars Discord.

The solution you posted is affected by a series of beginner errors and it will be not easy to explain all of them here. For example, this line has more than one mistake in it: `if(Number.isInteger("i")&& (i <= 10 && i <= 0)){`.

I am closing this discussion as off-topic here. Please ask your question in kata discourse, or on CW Discord.
